akka-stream/src/main/scala/akka/stream/stage/Stage.scala

```diff
@@ -306,6 +310,125 @@ abstract class AsyncStage[In, Out, Ext]
 }
 
 /**
+ * INTERNAL API
+ */
+private[stream] object TimerStage {
+  final case class Scheduled(timerKey: Any, timerId: Int, repeating: Boolean) extends DeadLetterSuppression
+
+  sealed trait Queued
+  final case class QueuedSchedule(timerKey: Any, interval: FiniteDuration) extends Queued
+  final case class QueuedScheduleOnce(timerKey: Any, delay: FiniteDuration) extends Queued
+  final case class QueuedCancelTimer(timerKey: Any) extends Queued
+
+  final case class Timer(id: Int, task: Cancellable)
+}
+
+/**
+ * This is a variant of [[DetachedStage]] that can receive timer events.
+ *
+ * When overriding the preStart() and postStop() methods make sure that you call
+ * the inherited methods as the timer support relies on these.
+ */
+abstract class TimerStage[In, Out] extends AsyncStage[In, Out, TimerStage.Scheduled] {
```

Comment:
The current TimerStage is perhaps too difficult to use for the casual user due to the AsyncStage so it's not a good end user api for the 90% cases anyway. I get the impression that GraphStage is more flexible and should therefore be our low level building block, and then we can add something on top of that specially targeted for the simple/limited use cases.